app/controllers/badges_controller.rb

```diff
@@ -1,6 +1,7 @@
 class BadgesController < ApplicationController
   before_action :authenticate_user!
   before_action :set_badge, only: [:show]
+  load_and_authorize_resource except: [:require, :redeem]
```

Comment:
@frmendes I'm gonna merge this to master due to the short amount of time we have left, and because it is Sunday, it's not healthy for you to review PRs on Sundays!
Anyway, although this will be closed I would still like to continue discussing this :)